subprojects/docs/src/docs/userguide/authoring-builds/basics/writing_plugins.adoc

```diff
@@ -0,0 +1,207 @@
+// Copyright (C) 2023 Gradle, Inc.
+//
+// Licensed under the Creative Commons Attribution-Noncommercial-ShareAlike 4.0 International License.;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://creativecommons.org/licenses/by-nc-sa/4.0/
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[writing_plugins]]
+= Writing Plugins
+
+Writing custom plugins is the best way to customize and organize your build logic.
+
+== Sample plugin
+
+A plugin is any class that implements the link:{javadocPath}/org/gradle/api/Plugin.html[`Plugin`] interface.
+
+The example below is the most straightforward plugin, a "hello world" *plugin*:
+
+.build.gradle.kts
+[source,kotlin]
+----
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+
+abstract class SamplePlugin : Plugin<Project> {
+    override fun apply(project: Project) {
+        project.tasks.create("SampleTask") {
+            println("Hello world!")
+        }
+    }
+}
+----
+
+== Script plugin
+
+The simplest plugin is a script plugin coded in the build script.
```

Comment:
💭 A little weird to talk about the "simplest" plugin, when the previous example was simpler.  I think talking about subclassing `DefaultTask` is unnecessary, maybe this could use a built-in task?